EasyPost/Exceptions/API/_Base.cs

```diff
@@ -73,6 +74,13 @@ internal static ApiError FromErrorResponse(RestResponse response)
         {
             // NOTE: This method anticipates that the status code will be a non-2xx code.
             // Do not use this method to parse a successful response.
+
+            // short-circuit if the request timed out
+            if (response.ErrorException?.GetType() == typeof(TaskCanceledException))
```

Comment:
question: is this check robust enough to catch all timeouts? Meaning, I know that for this particular use-case it triggered a `TaskCanceledException`, but will that always be the case?